src/codeManager.ts

```diff
@@ -54,7 +54,15 @@ export class CodeManager implements vscode.Disposable {
         this.initialize();
 
         const fileExtension = extname(this._document.fileName);
-        const executor = this.getExecutor(languageId, fileExtension);
+        let executor = null;
+
+        const firstLineInFile = this._document.lineAt(0).text;
+        if (firstLineInFile.startsWith("#!")) {
+            executor = firstLineInFile.substr(2);
+        } else {
+            executor = this.getExecutor(languageId, fileExtension);

```

Comment:
If languageId is set (i.e. when it is called by `runByLanguage()`), we should not use shebang.

Reply:
Added check for null